src/nvim/lua/vim.lua

```diff
@@ -230,31 +231,42 @@ paste = (function()
     end
     return true  -- Paste will not continue if not returning `true`.
   end
-end)()
+end
 
 --- Defers callback `cb` until the Nvim API is safe to call.
 ---
 ---@see |lua-loop-callbacks|
 ---@see |vim.schedule()|
 ---@see |vim.in_fast_event()|
-local function schedule_wrap(cb)
+function vim.schedule_wrap(cb)
   return (function (...)
     local args = {...}
     vim.schedule(function() cb(unpack(args)) end)
   end)
 end
 
+vim._update_package_paths = _update_package_paths
+vim._os_proc_children = _os_proc_children
+vim._os_proc_info = _os_proc_info
+vim._system = _system
```

Comment:
why do this here in stead of where they're defined? if it's to trick luacheck, just add a `-- luacheck: ...` override.

Reply:
It's because those functions referenced eachother, and I didn't see a need to change it.

Reply:
you mean to avoid search/replace churn? Would be better to do the search-replace, to avoid people accidentally copying this pattern.